pandas/core/frame.py

```diff
@@ -4831,36 +4832,41 @@ def corrwith(self, other, axis=0, drop=False):
             0 or 'index' to compute column-wise, 1 or 'columns' for row-wise
         drop : boolean, default False
             Drop missing indices from result, default returns union of all
+        method : {'pearson', 'kendall', 'spearman'}
+            * pearson : standard correlation coefficient
+            * kendall : Kendall Tau correlation coefficient
+            * spearman : Spearman rank correlation
+
+            .. versionadded:: 0.20.0
+
+        min_periods : int, optional
+            Minimum number of observations needed to have a valid result
+
+            .. versionadded:: 0.20.0
 
         Returns
         -------
         correls : Series
         """
         axis = self._get_axis_number(axis)
         if isinstance(other, Series):
-            return self.apply(other.corr, axis=axis)
+            return self.apply(other.corr, axis=axis,
+                              method=method, min_periods=min_periods)
 
         this = self._get_numeric_data()
         other = other._get_numeric_data()
 
         left, right = this.align(other, join='inner', copy=False)
 
-        # mask missing values
-        left = left + right * 0
-        right = right + left * 0
-
         if axis == 1:
             left = left.T
             right = right.T
 
-        # demeaned data
-        ldem = left - left.mean()
-        rdem = right - right.mean()
-
-        num = (ldem * rdem).sum()
-        dom = (left.count() - 1) * left.std() * right.std()
-
-        correl = num / dom
+        correl = Series({col: nanops.nancorr(left[col].values,
+                                             right[col].values,
```

Comment:
I think the new implementation (which calls nancorr which calls numpy/scipy correlation functions) is actually significantly faster than the current implementation (manually computing Pearson correlation using `DataFrame.mean()`, `DataFrame.sum()`, and `DataFrame.std()`)

For example:

Current implementation:
```python
>>> import pandas as pd; import timeit
>>> pd.__version__
u'0.19.2'
>>> iris = pd.read_csv('https://raw.githubusercontent.com/mwaskom/seaborn-data/master/iris.csv')
>>> timeit.timeit(lambda: iris.corrwith(iris), number=10000)
50.891642808914185
>>> timeit.timeit(lambda: iris.T.corrwith(iris.T), number=10000)
42.0677649974823
```

New implementation:
```python
>>> import pandas as pd; import timeit
>>> pd.__version__
'0.19.0+539.g0b77680.dirty'
>>> iris = pd.read_csv('https://raw.githubusercontent.com/mwaskom/seaborn-data/master/iris.csv')
>>> timeit.timeit(lambda: iris.corrwith(iris, method='pearson'), number=10000)
28.622286081314087
>>> timeit.timeit(lambda: iris.T.corrwith(iris.T, method='pearson'), number=10000)
21.898916959762573
```

I'm pretty new to this, so please let me know if I'm missing anything here.